internal/controller/provisioner/objects.go

```diff
@@ -160,7 +163,25 @@ func (p *NginxProvisioner) buildNginxResourceObjects(
 	if p.isOpenshift {
 		objects = append(objects, openshiftObjs...)
 	}
-	objects = append(objects, service, deployment)
+
+	if nProxyCfg.Kubernetes.Deployment.Autoscaling.Enabled {
```

Comment:
there's need to be a nil check for Autoscaling here

```suggestion
if nProxyCfg.Kubernetes.Deployment.Autoscaling != nil {
	if nProxyCfg.Kubernetes.Deployment.Autoscaling.Enabled {
```